src/realm/sync/noinst/protocol_codec.hpp

```diff
@@ -304,8 +304,11 @@ class ClientProtocol {
                             info.compensating_writes.push_back(std::move(cwei));
                         }
 
-                        info.compensating_write_server_version =
-                            json.at("compensatingWriteServerVersion").get<int64_t>();
+                        if (auto server_version = json.find("compensatingWriteServerVersion");
+                            server_version != json.end()) {
+                            info.compensating_write_server_version =
+                                std::make_optional<version_type>(server_version->get<int64_t>());
+                        }
```

Comment:
Add a comment that this value is not provided for Asymmetric sync compensating writes

Reply:
I don't know if that would be correct. I think this value is not provided if there is a permission issue, for example. cc @kmorkos

Reply:
This value is not provided when the write_not_allowed(230) error is received for Asymmetric Sync/Data Ingest. (I misspoke when I said compensating writes).